clang/test/BoundsSafety/Sema/attributes_in_template_decls.cpp

```diff
@@ -0,0 +1,95 @@
+// RUN: %clang_cc1 -fsyntax-only -fbounds-safety -fexperimental-bounds-safety-cxx -verify %s
+// This test cases causes an assertion failure. Consequently the `expected-*` lines
+// are probably wrong. They can't be fixed until the compiler can parse this code without
+// hitting an assert (rdar://152452705).
```

Comment:
What happens if the assert is removed? Also, we should describe the issue in more detail than just "hitting an assert", such that people without radar access can figure out why the test case is disabled.

Reply:
I didn't think it was that important. It's the experimental C++ mode that no-one is using. I can add more description on the assert.

Reply:
I've added a better description of the assert and also made the test pass if the assert is removed.

I don't think it makes sense to update the assert because the code in `diagnoseDynamicCountVarInit` is wrong for C++ and removing the assert will just hide this.